IlmBase/IlmThread/IlmThread.h

```diff
@@ -94,19 +94,25 @@
 #include "IlmThreadExport.h"
 #include "IlmThreadNamespace.h"
 
-#ifdef ILMBASE_FORCE_CXX03
-#   if (defined (_WIN32) || defined (_WIN64)) && !defined(HAVE_PTHREAD)
+#if ILMBASE_FORCE_CXX03
+#   if defined (_WIN32) || defined (_WIN64)
+#   if !HAVE_PTHREAD
 #       ifdef NOMINMAX
 #          undef NOMINMAX
 #       endif
 #       define NOMINMAX
 #       include <windows.h>
 #       include <process.h>
-#   elif HAVE_PTHREAD
-#      include <pthread.h>
+#   else
+#      include <pthread.h>
+#   endif
+#   endif
+#else
+#   if defined(__MINGW32__) || defined(__MINGW64__)
```

Comment:
I am asking for your understanding. I'm just an amateur.
In my opinion, MINGW makes sense here because the definition of std :: thread is used below. Looks worse adding MINGW for std :: mutex under windows for gcc as below is cut. Struct semaphore {} works badly.

PS. New definition std::jthread has appeared. Apparently more modern than std::thread. For c++17 I managed to compile. However for c++11 I couldn't find the jthread.h plugin.

Reply:
jthread is in c++20, which is beyond our target. This change makes sense, my question was if the main topic of your PR is to select the threading api with more detail than the existing code.